### PR TITLE
initial support for Apache Cassandra

### PR DESCRIPTION
**Pull request by mabi**
Experimental development for Apache Cassandra. Feed engine works. Table structure and code to be improved.

### CODE REVIEW

File: Modules/feed/engine/CassandraEngine.php [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    /**
+     * Create feed
+     *
+     * @param integer $feedid The id of the feed to be created
+     * @param array $options for the engine
+     *
+     * {@inheritdoc}
+     *
+     */
+    public function create($feedid,$options)
```
Comment (TrystanLea) [inline Modules/feed/engine/CassandraEngine.php]: Can you add $feedid = (int) $feedid; here to help with input sanitisation?

Comment (TrystanLea) [inline Modules/feed/engine/CassandraEngine.php]: I've fixed in latest commit

**Comment by TrystanLea** [comment]
Thanks Mabi, I havent been able to test with Cassandra but I have tested to check it doesnt affect a standard emoncms system and quickly reviewed the engine code and added extra input sanitisation

**Comment by mabi** [comment]
Thank you @TrystanLea. I am testing the code with a single emonpi, so far so good. I will improve the code and implement missing features (eg cvs export). I think Cassandra is well suited for this kind of data.

**Pull request merged**